### ISSUE TITLE
Safe math is not safe

### ISSUE THREAD

**Issue by shawntabrizi** [parity-samples/substrate-erc20-multi#3]
https://github.com/parity-samples/substrate-erc20-multi/blob/0a6615c81204068996786e79ae05af97b7e7f51b/runtime/src/erc20.rs#L69

When the `if` statement fails, it will continue to run the rest of the code underneath.

You need to return an `Err` when the checked math fails so that the runtime execution halts. Like so:

https://github.com/shawntabrizi/substrate-collectables-workshop/blob/master/2/tracking-all-kitties.md#checking-for-overflowunderflow

```
let new_all_people_count = all_people_count.checked_add(1).ok_or("Overflow adding a new person")?;
```

**Comment by shawntabrizi** [comment]
You should also not be doing that check in the middle of a `mutate`. You need to follow the `Verify First, Write Last` pattern: https://github.com/shawntabrizi/substrate-collectables-workshop/blob/master/2/tracking-all-kitties.md#verify-first-write-last

**Comment by gautamdhameja** [comment]
Fixed in c5305b9a15b963529978862300b91d2720abeda1. Closing.
